diesel/src/query_builder/uncorrelated_column.rs

```diff
@@ -0,0 +1,55 @@
+use crate::backend::{Backend, DieselReserveSpecialization};
+use crate::query_builder::{AstPass, QueryFragment};
+use crate::{Column, Expression, QueryResult};
+
+/// Provide a wrapper type which will render a `Column` without its
+/// 'correlation' (i.e. table name).
+pub(crate) trait Uncorrelated {
+    /// The `Column` type to wrap.
+    type Column: Column;
+
+    /// `Column` wrapped to signify it shouldn't be rendered with its
+    /// correlation.
+    fn uncorrelated(self) -> UncorrelatedColumn<Self::Column>;
+}
```

Comment:
As far as I can tell, we don't need this trait at all. We can just all `AssignableExpression::into_assignable_expression` for the inner type of `ArrayIndex`.